### DISCUSSION TITLE
How to customize the font in the insert_textbox function

### DISCUSSION THREAD

**Discussion by TCHSDUFH** [pymupdf/PyMuPDF#3935] [Looking for help] [answered]
When using the insert_textbox function, I want to use some Chinese fonts, such as SimSum.ttf. I checked the source code of the function and found that it doesn't support this font. It's also not possible to directly assign a local font file path. How should I proceed to make the inserted text use the SimSum.ttf font? tks
```python
page.insert_textbox(rect,
                            nowPage,
                            fontsize=int(pageFontSet.fontSize),
                            fontname='D:\\SimSum.ttf',    
                            color=pageFontSet.fontColor,    
                            align=TEXT_ALIGN_CENTER,  
                            overlay=True)  # 
```

**Comment by JorjMcKie** [comment]
This is not an issue, but a Discussions item.

**Comment by JorjMcKie** [comment, marked as answer]
To support a font file or a font in memory, use the `page.insert_font()` method.
In general please note that the "fontname"  parameter **_is not the font filename!_**
This is a symbolic name (like a variable name) by which the font is referenced.
Therefore, do this:
```python
page.insert_font(fontname="myfont", fontfile='D:\\SimSum.ttf')  # or fontbuffer=...
page.insert_textbox(...., fontname="myfont")
```

**Reply by TCHSDUFH** [reply]
I saw issue #3438 and tried that solution. It has already solved my problem. Thank you very much for the response and guidance.
